### PR TITLE
refactor and avoid flashing main page on startup

### PR DESCRIPTION
**Pull request by pvuorela**

### CODE REVIEW

**Comment by pvuorela** [comment]
@rainemak & @dcaliste 

Found a crash too: pdf open and clicking another on transfers. But that already exists, so not a regression.

**Comment by pvuorela** [comment]
Forgot to add dbus service to .desktop file.

**Comment by pvuorela** [comment]
There.

File: main.cpp [hunk 1]
```diff
@@ -115,16 +98,29 @@ Q_DECL_EXPORT int main(int argc, char *argv[])
...
-    context->setContextProperty(QStringLiteral("applicationWindow"), view.data());
-
     //% "Documents"
     Q_UNUSED(QT_TRID_NOOP("sailfish-office-ap-name"))
 
+    bool preStart = false;
+    QString fileName;
+
+    for (int i = 1; i < argc; ++i) {
+        if (QString(argv[i]) == "-prestart") {
```
Comment (rainemak) [inline main.cpp]: Use ```QStringLiteral("-prestart")``` in the comparison (http://doc.qt.io/qt-5/qstring.html#QStringLiteral).

I'd prefer using app->arguments() QStringList but this does the job as well.

File: qml/Main.qml [hunk 2]
```diff
@@ -26,7 +26,90 @@ ApplicationWindow
     id: window
 
     property Item documentItem
-    property QtObject fileListModel
+    property QtObject fileListModel: trackerProvider.model
+    property Page _mainPage
```
Comment (rainemak) [inline qml/Main.qml]: Imo not needed to be a private property.

File: qml/Main.qml [hunk 3]
```diff
@@ -26,7 +26,90 @@ ApplicationWindow
...
+
+        default:
+            console.log("Warning: Unrecognised file type for file " + fileInfo.fullPath)
+        }
+
+        if (handler != "") {
+            pageStack.push(handler,
+                           { title: fileInfo.fileName, path: fileInfo.fullPath, mimeType: fileInfo.mimeType },
+                           PageStackAction.Immediate)
+        }
```
Comment (rainemak) [inline qml/Main.qml]: What happens if there is no handler? Maybe something should be shown for the user to indicate that this file type is unknown.

Comment (pvuorela) [inline qml/Main.qml]: Could, but need then to decide UI, have new strings and whatnot. Can create a task.

File: qml/Main.qml (hunk 2 above)
Comment (pvuorela) [inline qml/Main.qml]: I think I rather keep for now as rest are used from outside.

**Comment by pvuorela** [comment]
Added string literal

**Comment by rainemak** [comment]
App startup with a file takes quite a bit of hit with this as the first frame rendered only after everything is ready. The cost is calculated in seconds, when using qml profiler.

Normally first delegates of the FileListPage are created ~2.5secs after start. With this PR we are seeing the first frame >4secs after start.

Would it be possible to prepare the content the first page asynchronously in case we're launching office with a document file?

File: qml/Main.qml (hunk 3 above)
Comment (rainemak) [inline qml/Main.qml]: Thanks.

**Comment by pvuorela** [comment]
As commented offline, I think file open startup optimizations should be handled separately.

**Comment by rainemak** [comment]
LGTM

**Pull request merged**